src/Bridge/Doctrine/DBAL/Types/TypesDumper.php

```diff
@@ -74,14 +80,47 @@ private function dump(array $types): string
         return $code;
     }
 
-    private function getTypeCode(string $classname, string $enumClass, string $type, string $name): string
-    {
+    private function getTypeCode(
+        string $classname,
+        string $enumClass,
+        string $type,
+        string $name,
+        ?string $defaultOnNull
+    ): string {
+        $implementation = <<<PHP
+            public const NAME = '$name';
+
+            protected function getEnumClass(): string
+            {
+                return \\{$enumClass}::class;
+            }
+
+            public function getName(): string
+            {
+                return static::NAME;
+            }
+PHP;
         switch ($type) {
             case self::TYPE_INT:
                 $baseClass = AbstractIntegerEnumType::class;
                 break;
             case self::TYPE_STRING:
                 $baseClass = AbstractEnumType::class;
+
+                if ($defaultOnNull !== null) {
+                    $implementation .= <<<PHP
+
+            protected function onNullFromDatabase()
+            {
+                return \\{$defaultOnNull};
```

Comment:
As the configuration will contain the enum raw value, we should get an enum instance from it here using `EnumInterface::get`

Reply:
I'll give this a try but actually I tested this with an enum name like `App\Enum\MyEnum::FOO`.

I didn't think that people would want to put raw values into their configuration since that would mean having to change the configuration on enum value change.

Reply:
Yes of course, either using `App\Enum\MyEnum::FOO` or the value directly, it's the same. That's what I meant :)

Reply:
OK, I've changed this now. I still needed to "unpack" the enum via `->getValue()` but that is to be expected.